refactor based on new common gamescope-session

alkazar opened the pull request:
Based on changes to gamescope-session here: https://github.com/ChimeraOS/gamescope-session/pull/64

Essentially allows sharing of almost all of the session scripts - no more need for copying.
Should be much easier to maintain this way :)

alkazar committed:
refactor based on new common gamescope-session

pastaq commented on usr/share/gamescope-session/sessions.d/steam-plus:
```diff
@@ -0,0 +1,7 @@
+[[ -f "${STEAM_SESSION_FILE}" ]] && . "${STEAM_SESSION_FILE}"
+
+CURSOR_FILE="${HOME}/.local/share/Steam/tenfoot/resource/images/cursors/arrow_right.png"
+CLIENTCMD="opengamepadui --only-qam -- $CLIENTCMD"
```
Is the intention here to concatenate the historical STEAMCMD and OGUICMD? The second half of this needs to be the command for the underlay process that OpenGamepadUI will launch. This is typically steam and all of its arguments. Previously we separated this into two variables. If you echo $CLIENTCMD here will it be "opengamepadui --only-qam -- steam -steamos3 -etc"?

pastaq replied:
Also, we're going to change the name from --only-qam so we should update that now and merge everything at the same time.

ShadowApex replied:
Yes, this would source the whole steam gamescope session, and execute the `CLIENTCMD` that is defined in the steam session.

alkazar replied:
Yes, it works like ShadowApex explained it. Is that acceptable?

The common gamescope session script now expects the `CLIENTCMD` variable. This way we can reuse the steam session for steam-plus.

ShadowApex replied:
The only thing I'm not a fan of is that this would make all of opengamepadui-session (as a package) dependent on the steam session, when it's really only necessary for the "steam-plus" session. And if we choose to not make the steam session a hard dependency, then users who try to use this session would find it broken.

With that in mind, I think our options are:
* Make the steam session a hard dependency ( :cry: )
* Source from the steam session if it exists, but fallback to a functional default
* Create a separate "steam-plus" session package/repo that has a dependency on both the steam and opengamepadui packages.

alkazar replied:
This was already the case before. I think we can just make it an optional dependency. If steam/steam session is not available, the steam-plus session will currently just fail. I am not sure what alternative behaviour we could have.

pastaq replied:
> This was already the case before. I think we can just make it an optional dependency. If steam/steam session is not available, the steam-plus session will currently just fail. I am not sure what alternative behaviour we could have.

Not exactly. Before it was able to pick up a custom STEAMCMD from a .config/environment.d conf, but the default STEAMCMD was explicit in the session script.

We probably don't want to pull from the steam session script anyway as it ensures we will always need the same steam arguments for both sessions, which we're working towards removing as many as possible in the Plus session.

pastaq replied:
Can we adjust this?

ShadowApex commented:
With this change, we should note that we will need to change the systemd-extension packaging to include both the opengamepadui-session and the dependent gamescope-session. Otherwise this will break that installation method (most notably on the Steam Deck + SteamOS)

ShadowApex commented on usr/share/wayland-sessions/gamescope-session-opengamepadui.desktop:
Because this change renames the session `.desktop` file, we'll also need to update `opengamepadui-session-select` with the correct filename:
https://github.com/ShadowBlip/OpenGamepadUI-session/blob/main/usr/bin/opengamepadui-session-select#L72-L74

This session select script is only used on SteamOS platform installations.

alkazar replied:
I was thinking about this, and maybe we keep the old names for compatibility? I was going for naming consistency, but that isn't so important.

ShadowApex replied:
I think it's ok to rename the desktop files for consistency. We just also need to update the session select script to use the new name.

alkazar replied:
well, the other issue is i think it will break existing installs unless we do some kind of migration to adjust user's settings.

ShadowApex replied:
That's very true. Let's also keep the old desktop file for compatibility, but also use the new one. We can deprecate the old one and write a migration for it in the future.

alkazar committed:
update for gamescope-session-plus

alkazar committed:
various fixes

 - update session switcher script with new session names
 - include symlinks of old session names for compatibility

pastaq commented on usr/share/gamescope-session-plus/sessions.d/steam-plus:
```diff
@@ -0,0 +1,7 @@
+[[ -f "${STEAM_SESSION_FILE}" ]] && . "${STEAM_SESSION_FILE}"
+
+CURSOR_FILE="${HOME}/.local/share/Steam/tenfoot/resource/images/cursors/arrow_right.png"
+CLIENTCMD="opengamepadui --only-qam -- $CLIENTCMD"
```
There is a  problem with this. We depend on the steam session, which is not what we want. It is very likely we will need different arguments in the future that will be incompatible with the standalone session.

I also have a question, how will users override the client command in . config if we are sourcing from the steam script?

alkazar replied:
Depending on the steam session is exactly what we want for the steam big picture plus session. We need all the environment that is setup there for Steam. If we need to alter the arguments we can still do that.

If users override `CLIENTCMD` it will override the entire contents of the variable.

Also, it is probably better to create your own session files instead of overriding `CLIENTCMD` at this point.

alkazar commented:
I think this is good to merge. gamescope-session and gamescope-session-steam PRs have been merged.
